pkg/compiler/analysis.go

```diff
@@ -143,26 +147,30 @@ func (c *codegen) traverseGlobals() bool {
 // countGlobals counts the global variables in the program to add
 // them with the stack size of the function.
 // Second returned argument contains the amount of global constants.
-func countGlobals(f ast.Node) (int, int) {
-	var numVar, numConst int
+func countGlobals(f ast.Node) (int, int, int) {
+	var numVar, numUnusedVarWithCall, numConst int
 	ast.Inspect(f, func(node ast.Node) bool {
 		switch n := node.(type) {
 		// Skip all function declarations if we have already encountered `defer`.
 		case *ast.FuncDecl:
 			return false
 		// After skipping all funcDecls, we are sure that each value spec
-		// is a global declared variable or constant.
+		// is a globally declared variable or constant.
 		case *ast.GenDecl:
 			isVar := n.Tok == token.VAR
 			if isVar || n.Tok == token.CONST {
 				for _, s := range n.Specs {
-					for _, id := range s.(*ast.ValueSpec).Names {
+					valueSpec := s.(*ast.ValueSpec)
+					multiRet := len(valueSpec.Values) != 0 && len(valueSpec.Names) != len(valueSpec.Values) // e.g. var A, B = f() where func f() (int, int)
+					for j, id := range s.(*ast.ValueSpec).Names {
 						if id.Name != "_" {
 							if isVar {
 								numVar++
 							} else {
 								numConst++
 							}
+						} else if isVar && len(valueSpec.Values) != 0 && (j == 0 || !multiRet) && containsCall(valueSpec.Values[j]) {
+							numUnusedVarWithCall++
```

Comment:
Nice catch, it wasn't expected, fixed. Current implementation doesn't count unnamed vars with call inside, because we don't need it. Instead it only checks for their presence.